src/clr-angular/forms/combobox/model/multi-select-combobox.model.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ComboboxModel } from './combobox.model';
+
+export class MultiSelectComboboxModel<T> implements ComboboxModel<T> {
+  model: T[];
```

Comment:
There are some direct assignments in test, which can be refactored, but the currently _legal_  way to assign it is through the `OptionSelectionService`. Not sure how we can implement it without exposing a setter too.

Reply:
You can have a setter without a getter. I would prefer that the tests use the _legal_ way :).

Reply:
I'm sorry, but it seems that we have much more cases that need access to the model and I don't see how we can avoid this:
- OptionSelectionService.setSelectionValue() has checks if we are not setting the same model.
- Combobox.ts
   - Multiple model accesses, tied to the ControlValueAccessor implementation
   - User in logic determining display name
   - Used in prepopulation of the search text with the 1st selected item
   - Used in retrieving the id of the active-descendant for the keyboard navigation
- FocusHandler
   - Retrieving the index of the currently focused item
   - ScrollIntoView functionality for currently focused item
   - Retrieving and focusing first selected item in a pre-selected combobox 
If we add both getter and setter - then it's no different from just a plain public property.

Reply:
My main point is that there isn't much value in the rest of this class if you can directly modify the model, as its really just providing helper methods instead of managing the model. If someone sets the model to an empty array instead of a null, then we get into a strange state for validation, as one example. The reason I'm focusing on this is that the model is the most likely place we'll get something out of sync between user interactions.

Reply:
I understand your point, and yes - currently it is just a model holder with utility methods.
And my point is that, as long as we implement ControlValueAccessor, we're required to allow direct access. The only meaningful action item I can think of, is to introduce setter and getter and force null in the case of empty array in the setter.
If you have any other actionable ideas, please advise!
I also can't really advocate why it is designed as it is, I don't even know who designed it initially. But it's buried deep inside the combobox architecture and if we decide that it's important to do it differently, we should be aware that it may take a bit more time, as it will affect many aspects of the Combobox implementation that we currently have. If we want another architecture, we should talk about it and define its strict requirements.

Reply:
Alright, let's then add a comment about this as a TODO for this.